src/event.rs

```diff
@@ -262,179 +262,123 @@ pub enum Key {
 
 impl Key {
     #[cfg_attr(any(feature = "test", test), allow(dead_code))]
-    fn from_keycode(key: glutin_event::VirtualKeyCode) -> Option<Self> {
-        use glutin_event::VirtualKeyCode::*;
+    fn from_keycode(key: VirtualKeyCode) -> Option<Self> {
         Some(match key {
```

Comment:
```suggestion
        #[deny(unreachable_patterns, unused_variables)]
        Some(match key {
```

Could you change this line to include this attribute instead of prefixing every variant with the enum name? This fixes the lack of compilation errors without forcing us to repeat that name on every line.

Reply:
I don't think that is a good idea: having the name enables the compiler to say "at that exact line there is a problem because [...]" and not "around this huge block of dozens of lines, there is something weird", in the latter case we would just loose time for no reason.

Reply:
@PaulDance These errors will be quite rare (only when keycodes change), so I'd much rather optimize for having less code to read through than the precision of the error message. The main problem being solved here is that the errors were "silent" (as warnings) and we would rather that they be actual errors. I'd really prefer to keep this code as it was before while solving only that problem.